src/serverExplorer.ts

```diff
@@ -115,12 +115,23 @@ export class ServersViewTreeDataProvider implements TreeDataProvider< Protocol.S
         view.reveal(data, { focus: true, select: true });
     }
 
-    public addDeployment(server: Protocol.ServerHandle): Thenable<Protocol.Status> {
+    public async addDeployment(server: Protocol.ServerHandle): Promise<Protocol.Status> {
+        // quickpick to solve a vscode api bug in windows that only opens file-picker dialog either in file or folder mode
+        const isWindows: boolean = process.platform.indexOf('win') >= 0;
+        let filePickerType = 'file or exploded';
+        if (isWindows) {
+            filePickerType = await window.showQuickPick(['File', 'Exploded'], {placeHolder:
+                'Which deployment do you want to add?'});
+            if (!filePickerType) {
+                return;
+            }
+        }
+
         return window.showOpenDialog({
-            canSelectFiles: true,
+            canSelectFiles: (isWindows ? filePickerType === 'File' : true),
             canSelectMany: false,
-            canSelectFolders: true,
-            openLabel: 'Select Deployment file or folder'
+            canSelectFolders: (isWindows ? filePickerType === 'Exploded' : true),
```

Comment:
I'd also extract the string 'Exploded' to an enum (?) and reuse it for the quickpick so that you dont have to repeat yourself.